diff --git a/CMakeLists.txt b/CMakeLists.txt
index cfee8efb..18dbf2c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,15 +144,9 @@ set_target_properties(tf2_py PROPERTIES
 #   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
 # )
 
-if(WIN32)
-  install(FILES ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_PYTHON_DESTINATION}/_tf2.pyd
-    DESTINATION ${CATKIN_PACKAGE_PYTHON_DESTINATION}
-  )
-else()
-  install(FILES ${CATKIN_DEVEL_PREFIX}/${CATKIN_PACKAGE_PYTHON_DESTINATION}/_tf2.so
+install(FILES $<TARGET_FILE:tf2_py>
     DESTINATION ${CATKIN_PACKAGE_PYTHON_DESTINATION}
-  )
-endif()
+)
 
 #############
 ## Testing ##
