diff --git a/CMakeLists.txt b/CMakeLists.txt
index af6bc62c..cb01537e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -75,10 +75,17 @@ set_target_properties(${PROJECT_NAME}_static_transform_publisher
   PROPERTIES OUTPUT_NAME static_transform_publisher
 )
 
-# Install rules
+# Install library
 install(TARGETS
-  ${PROJECT_NAME} ${PROJECT_NAME}_buffer_server
-  ${PROJECT_NAME}_static_transform_publisher
+  ${PROJECT_NAME}
+  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+  RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION}
+)
+
+# Install executable
+install(TARGETS
+  ${PROJECT_NAME}_buffer_server ${PROJECT_NAME}_static_transform_publisher
   ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
   RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
